### PR TITLE
Mention related MRs in MR descriptions

### PR DESCRIPTION
**Pull request by dancysoft**
When more than one commit is pushed at once, update the description of
all associated MRs to include a list of related MRs.  This makes it
easy to see what MRs are part of the same unit of work.

Sample description:

Related MRs:

* !23 (This MR)
* !25
* !24
* _production_

### CODE REVIEW

**Comment by dancysoft** [comment]
Example at https://gitlab.wikimedia.org/dancy/puppet/-/merge_requests/23

File: gerritlab/main.py [hunk 1]
```diff
@@ -96,6 +96,25 @@ def timing(timer_name):
...
         timers[timer_name] = timers.get(timer_name, 0) + elapsed
     
+def generate_augmented_mr_description(commits_data, commit):
+    if len(commits_data) <= 1:
+        # No augmentation if only push a single commit
+        return utils.get_msg_title_description(commit.commit.message)
+    
+    target_branch = commits_data[0].target_branch
+
+    extra = ["These changes were pushed together:"]
```
Comment (yaoyuannnn) [inline gerritlab/main.py]: How about `Related changes` (or `Related MRs`), just like in Gerrit?

File: gerritlab/main.py [hunk 2]
```diff
@@ -176,6 +186,23 @@ def create_merge_requests(repo, remote, local_branch):
                 mr.create()
             new_mrs.append(mr)
     
+    # At this point we have one MR for each commit.
+    # title/desc/target_branch may be out-of-date for preexisting MRs.
+    # Augment the MR descriptions to include a list of related MRs.
+    for c in commits_data:
+        title, desc = generate_augmented_mr_description(commits_data, c)
+
```
Comment (yaoyuannnn) [inline gerritlab/main.py]: Remove blank line?

**Review by yaoyuannnn — APPROVED** [review]
Thanks for implementing this much-needed feature! Much appreciated!

File: gerritlab/main.py (hunk 2 above)
Comment (dancysoft) [reply gerritlab/main.py]: Will do.

File: gerritlab/main.py (hunk 1 above)
Comment (dancysoft) [reply gerritlab/main.py]: I'll go with `Related MRs`.

**Review by thcipriani — APPROVED** [review]
Love it! Works great!

**Pull request merged**